backends/candle/src/lib.rs

```diff
@@ -188,7 +189,7 @@ impl CandleBackend {
         let device = if candle::utils::cuda_is_available() {
             #[cfg(feature = "cuda")]
             match compatible_compute_cap() {
-                Ok(true) => Device::new_cuda(0),
+                Ok(true) => Device::new_cuda(device_id),
```

Comment:
IMO we should try to add some sort of validation here, to ensure that the `device_id` is within the bounds of the available devices in the given instance, or is the `candle` error enough?

Reply:
I feel like the candle error would be descriptive enough.